gpflow/utilities/bijectors.py

```diff
@@ -5,17 +5,31 @@
 from .. import config
 from .utilities import to_default_float
 
-__all__ = ["positive", "triangular"]
-
 
-def positive(lower: Optional[float] = None):
-    lower_value = config.default_positive_minimum()
-    if lower_value is None:
-        return tfp.bijectors.Softplus()
-
-    shift = to_default_float(lower_value)
-    bijectors = [tfp.bijectors.AffineScalar(shift=shift), tfp.bijectors.Softplus()]
-    return tfp.bijectors.Chain(bijectors)
+__all__ = ["positive", "triangular"]
 
 
-triangular = tfp.bijectors.FillTriangular
+def positive(lower: Optional[float] = None,
+             base: Optional[tfp.bijectors.Bijector] = None) -> tfp.bijectors.Bijector:
+    """
+    Returns a positive bijector (a reversible transformation from real to positive numbers).
+
+    :param lower: lower bound override (defaults to config.default_positive_minimum())
+    :param base: overrides the default base positive bijector (defaults to config.default_positive_bijector())
+    :returns: a bijector instance
+    """
+    bijector = base if base is not None else config.default_positive_bijector()
+    if lower is None:
+        lower = config.default_positive_minimum()
+    if lower is not None:
+        # Apply lower bound shift after applying base positive bijector
+        shift = tfp.bijectors.AffineScalar(shift=to_default_float(lower))
+        bijector = tfp.bijectors.Chain([bijector, shift])
+    return bijector
+
+
+def triangular() -> tfp.bijectors.Bijector:
+    """
+    Returns instance of a triangular bijector.
+    """
+    return tfp.bijectors.FillTriangular()
```

Comment:
Is it wrong to re-use bijectors ?
In that case, the code in `__config__.py` needs changing, too, as there the config's `positive` bijector is already the instance, not the type (`default_positve_bijector()` returns the config value, doesn't call it).

Reply:
Yes, it is wrong to re-use bijectors. They have cache inside.

Reply:
Then have to change the __config__ code, too.